sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java

```diff
@@ -98,6 +106,8 @@ public void startBroker() throws Exception {
 
     // create JMS connection factory
     connectionFactory = new ActiveMQConnectionFactory(BROKER_URL);
+    connectionFactoryWithoutPrefetch =
+        new ActiveMQConnectionFactory(BROKER_URL + "?jms.prefetchPolicy.all=0");
```

Comment:
No, it's just to guarantee that the reader get all messages without delay: ActiveMQ is pushing the messages to a cache (prefetch) before pushing to the client. In the case of the test, we don't want any delay between production and consumption (as we use `receiveNoWait()`). That's why I don't use prefetch just in the case of the test (the end user can use prefetch without problem).